src/analog/m2kanalogout_impl.cpp

```diff
@@ -598,3 +599,13 @@ unsigned int M2kAnalogOutImpl::getNbChannels()
 {
 	return m_dac_devices.size();
 }
+
+string M2kAnalogOutImpl::getChannelName(unsigned int channel)
+{
+	std::string name = "";
+	auto chn = getDacDevice(channel)->getChannel(0, true);
+	if (chn) {
+		name = chn->getName();
```

Comment:
The name of the iio_channel is empty. I think you want to retrieve its id (e.g. voltage0).